src/Form/SentencesSearchForm.php

```diff
@@ -268,6 +307,33 @@ public function setData(array $data)
         }
     }
 
+    private function paramIndex($param) {
+        $index = array_search($param, $this->paramsOrder);
+        if ($index === FALSE) {
+            $index = PHP_INT_MAX;
+        }
+        return $index;
+    }
+
+    public function getData($field = null) {
+        $data = parent::getData($field);
+        if (is_null($field)) {
+            uksort($data, function($a, $b) {
+                return $this->paramIndex($a) <=> $this->paramIndex($b);
+            });
+        }
+        return $data;
+    }
+
+    public function generateRandomSeedIfNeeded() {
```

Comment:
Should the check for the `rand_seed` parameter only work for the pagination links or also for the sidebar search form?

I.e. if I go to the advanced search page and set a query string, a language and random order, I get redirected to the result page and the search form in the sidebar is prefilled with the search settings. If I then change another parameter but keep the random order should I get the result in the same random order or not?

If yes, then that currently doesn't work because the advanced form never sets the random seed. (You would need to set a hidden field in the form.)

If no, I wonder whether it makes sense to have the option for the reversed order when the current sort order is random. After submitting the form, a new random seed will be generated and so the result isn't in reversed order but in a new random order.

Reply:
> I then change another parameter but keep the random order should I get the result in the same random order or not?

Currently, you can only get a new random order. Maybe I can add a new option in the sort dropdown: "Random (keep current order)" and "Random (generate new order)". These options would only show up in the side bar of a randomly sorted search.

> I wonder whether it makes sense to have the option for the reversed order when the current sort order is random

This is already a problem and this PR does not aim at solving it.